The initial version of the Topic Modeling modules (v0.1.0)

keamanansiber opened the pull request:
The initial version of the Topic Modeling modules (v0.1.0), contributed by Hatma Suryotrisongko, consists of:
- maryam/core/util/iris/topic.py
- maryam/modules/iris/topicmodeling.py

TESTED USING =
topicmodeling -i mixed.json -t json -m all-distilroberta-v1
topicmodeling -i mixed.json -t json -s -v -m all-distilroberta-v1
topicmodeling -i mixed.json -t json -s -m all-distilroberta-v1
topicmodeling -i mixed.json -t json -v -m all-distilroberta-v1
topicmodeling -i testdataset.csv -t csv -m all-mpnet-base-v2
topicmodeling -i testdataset.csv -t csv -s -v -m all-mpnet-base-v2
topicmodeling -i testdataset.csv -t csv -s -m all-mpnet-base-v2
topicmodeling -i testdataset.csv -t csv -v -m all-mpnet-base-v2

Note: download the dataset for testing from https://github.com/keamanansiber/Maryam/tree/master/notebooks

Commit:
The initial version of the Topic Modeling modules (v0.1.0), contributed by Hatma Suryotrisongko, consists of:
- maryam/core/util/iris/topic.py
- maryam/modules/iris/topicmodeling.py

saeeddhqan commented:
I'll review the pr when there is no change.

k4u5h1k commented:
Another error I got was this
<img width="525" alt="image" src="https://user-images.githubusercontent.com/59250093/179341626-7b39aba8-6213-48d0-869f-79b27d2f72ca.png">
This is because we have a script named `plotly.py` in the iris directory and python seems to think this is the plotly module. Please go ahead and rename it.

keamanansiber committed:
Merge branch 'saeeddhqan:master' into TopicModeling_0.1.0

keamanansiber committed:
Adding the --output and --api functionality. Tested by using =
topicmodeling -i mixed.json -t json -m all-distilroberta-v1  --output
report json testreport iris/topicmodeling
topicmodeling -i mixed.json -t json -m all-distilroberta-v1  --api

keamanansiber commented:
I have followed your suggestion, by renaming the plotly.py

In addition, I also added the --output and --api functionality. 
Tested by using =
topicmodeling -i mixed.json -t json -m all-distilroberta-v1  --output
report json testreport iris/topicmodeling
topicmodeling -i mixed.json -t json -m all-distilroberta-v1  --api

keamanansiber committed:
Adding Function: Search Topics Using a Keyword (Top2Vec)

saeeddhqan commented on maryam/modules/iris/topicmodeling.py:
```diff
@@ -0,0 +1,63 @@
+}
+
+
+def module_api(self):
```
show the output in the module_run function. In this function, just prepare the output.

saeeddhqan commented on maryam/modules/iris/topicmodeling.py:
```diff
@@ -0,0 +1,63 @@
+	print( results )
+
+	output = {'results': results.to_json(orient="records") }
+	print("\n\nOutput = \n")
```
use self.output instead of print

keamanansiber committed:
To answer Mr. Saeed's comments =
- show the output in the module_run function. In this function, just prepare the output.
- use self.output instead of print

saeeddhqan commented on maryam/core/util/iris/topic.py:
```diff
@@ -0,0 +1,225 @@
+# core/util/iris/topic.py
+# Hatma Suryotrisongko
+
+import pandas as pd
```
imports are heavy and decrease the speed of the framework. So, try to import them in the class.
also, add them all to the requirements file.

keamanansiber committed:
To answer Mr. Saeed's comments =
- imports are heavy and decrease the speed of the framework. So, try to import them in the class.
- also, add them all to the requirements file.
